lib/operationSpec.ts

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { HttpMethods } from "./webResource";
+
+/**
+ * A specification that defines an operation.
+ */
+export interface OperationSpec {
+  /**
+   * The HTTP method that should be used by requests for this operation.
+   */
+  httpMethod: HttpMethods;
```

Comment:
Just a thought: Do you want to have an index signature `[prop: string]: any;` so that someone can provide unknown/additional properties to the operation spec and pass it to the sendRequest method.

This could be useful while you are developing/designing the operation spec.

Reply:
To be brief, no, I don't think an index signature would be beneficial here.
I reserve index signatures for when an interface represents a map/dictionary type where the values don't mean anything to my code, but will mean something to my customer's code. In this scenario, OperationSpec is a set of properties that will populate a provided HttpRequest. If the customer added extra properties to the OperationSpec, my code in sendRequest() wouldn't know what those properties were, and even if it did, it wouldn't know what to do with them.